queue/queue.go

```diff
@@ -0,0 +1,125 @@
+package queue
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Generic represents a generic value.
+type Generic interface{}
+
+// Zero is the default zero value for a queue.
+var Zero Generic = nil
+
+// GenericQueue is a generic queue that is implemented as a linked list.
+type GenericQueue struct {
+	head *dNode
+	tail *dNode
+	len  int
+}
+
+// dNode is a node in a queue.
+type dNode struct {
+	data Generic
+	next *dNode
+	prev *dNode
+}
+
+// NewGenericQueue creates a new queue of generic values.
+func NewGenericQueue(vals ...Generic) *GenericQueue {
+	q := &GenericQueue{}
+
+	for _, v := range vals {
+		q.Push(v)
+	}
+
+	return q
+}
+
+func newDNode(data Generic) *dNode {
+	return &dNode{data: data}
+}
+
+func (q *GenericQueue) String() string {
+	if q.IsEmpty() {
+		return "[]"
+	}
+
+	sb := strings.Builder{}
```

Comment:
I updated how linked lists are printed in order to match slice printing. Update this to match that. There shouldn't be blank spaces between the brackets and elements.